spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -116,6 +119,16 @@ public boolean isStatsEnabled() {
 		return this.statsEnabled;
 	}
 
+	protected AbstractMessageChannelMetrics getMetrics() {
+		return this.channelMetrics;
+	}
+
+	@Override
+	public void configureMetrics(ConfigurableMetrics metrics) {
+		Assert.isInstanceOf(AbstractMessageChannelMetrics.class, metrics);
```

Comment:
Can't we do `ConfigurableMetricsAware` as generic?
````java
public interface ConfigurableMetricsAware<M extends ConfigurableMetrics> {

	void configureMetrics(M metrics);

}
````
And further:
````java
public abstract class AbstractMessageChannel implements ConfigurableMetricsAware<AbstractMessageChannelMetrics> {
....
@Override
public void configureMetrics(AbstractMessageChannelMetrics metrics) {
}
````
Because I don't see reason in this `isInstanceOf` if we really can require the appropriate type for method argument